### PR TITLE
TOOL-134 Added stack to bitrise-init.

### PR DESCRIPTION
**Pull request by richard-bogdan-bitrise**
The following PRs should be checked:
https://github.com/bitrise-io/bitrise-init/pull/164 (This)
https://github.com/bitrise-io/bitrise-add-new-project/pull/30
https://github.com/bitrise-io/bitrise/pull/673

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
 [[constraint]]
   name = "github.com/urfave/cli"
-  version = "1.20.0"
+  revision = "e6cf83ec39f6e1158ced1927d4ed14578fda8edb"
```
Comment (lszucs) [inline Gopkg.toml]: Please put a comment to clarify why we depend on specific revision.

I guess this is some dependency related error, but it is best to have it documented.

**Comment by lszucs** [comment]
@trapacska sorry, please wait until code is updated based on my comment. Will notify you again.

File: Gopkg.toml (hunk 1 above)
Comment (richard-bogdan-bitrise) [reply Gopkg.toml]: Added comment.

Reason: Urfave CLI uses go mod, and some dependencies were not resolved.

**Comment by lszucs** [comment]
@trapacska you may merge this now if no objections

File: scanner/utils.go [hunk 2]
```diff
@@ -211,7 +224,23 @@ func AskForConfig(scanResult models.ScanResultModel) (bitriseModels.BitriseDataM
 	}
 
 	config.App.Environments = append(config.App.Environments, appEnvs...)
+	config.Meta = map[string]interface{}{}
+	config.Meta["bitrise.io.stack"] = selectedStack
```
Comment (trapacska) [inline scanner/utils.go]: as far as I know, someone requested this to be changed, please make 100% sure this is what we need

File: stack/stack.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	for _, stack := range Stacks {
+		if string(stack) == s {
+			return stack, nil
+		}
+	}
+	return "", fmt.Errorf("could not find stack %s", s)
+}
+
+// StringValue returns the string value of the given Stack.
+func (s Stack) StringValue() string {
```
Comment (trapacska) [inline stack/stack.go]: This name should be `String` by Go's built-in.

File: stack/stack.go [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+}
+
+// Stack defines a Stack with it's name.
+type Stack string
+
+const (
```
Comment (trapacska) [inline stack/stack.go]: I think in a structure like os-tool-version might be better.
osx.xcode.112x
osx.vs4mac.stable

etc.

Will work like now but for me it is a bit of too much hardcoding for a relatively dynamic stuff.

File: stack/stack.go (hunk 3 above)
Comment (richard-bogdan-bitrise) [reply stack/stack.go]: Renamed.

File: stack/stack.go [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+	"osx-xcode-11.2.x",
+	"osx-xcode-10.0.x",
+	"osx-xcode-10.1.x",
+	"osx-xcode-10.2.x",
+	"osx-xcode-8.3.x",
+	"osx-xcode-9.4.x",
+	"osx-xcode-edge",
+}
+
+// StackOptionsMap holds the available stack for the given platforms, where the first item is the default.
```
Comment (lpusok) [inline stack/stack.go]: Should maintenance check for stack change check this struct? Comment left related to the PR: https://github.com/bitrise-io/bitrise-init/pull/174/files

File: stack/stack.go (hunk 4 above)
Comment (richard-bogdan-bitrise) [reply stack/stack.go]: Updated to what we discussed.

File: stack/stack.go (hunk 5 above)
Comment (richard-bogdan-bitrise) [reply stack/stack.go]: I guess it should. I added that ticket as a blocker.

**Comment by adborbas** [comment]
Hi @richard-bogdan-bitrise, 

I'm closing the PR as part of cleaning up old PRs. Please reopen if it is still valid. Thanks!

**Pull request closed without merging**